frontend/components/account/js/account-billing.component.ts

```diff
@@ -78,29 +79,47 @@ class AccountBillingController implements ng.IController {
 		return this.APIService.get(this.account + "/invoices")
 			.then((response) => {
 				this.billings = response.data;
+			})
+			.catch((error) => {
+				console.error(error);
 			});
 	}
 
 	public initPlans() {
-		return this.APIService.get("plans")
+		return this.APIService.get("/plans")
 			.then((response) => {
 				if (response.status === 200) {
+					this.billingError = false;
 					this.plans = response.data;
 				}
+			})
+			.catch((error) => {
+				this.billingError = true;
+				console.error(error);
 			});
 	}
 
 	public initSubscriptions() {
 		return this.APIService.get(this.account + "/subscriptions")
 			.then((response) => {
+				this.billingError = false;
 				this.subscriptions = response.data;
+			})
+			.catch((error) => {
+				this.billingError = true;
```

Comment:
all of these checks seems to overwrite  `billingError`.

let's say API request to `plan` is fetched before `subscriptions`... and `plan` failed but `subscriptions` succeed. `billingError` would then be set to false when it should be `true`.

I think this boolean needs to be set to `false` on init, and all these checks needs to be like written as something like:

````js
.then((response) => {
    this.billingError |= false; //so if the billingError was set to true already, it will stay true
    ...
})
.catch((error) => {
   this.billingError = true; //Here we don't need care about previous value, if it was false we want it to be true.
   ...
})
````

Alternatively, billingError should just be a function that returns `true` if any of the follows are undefined:
`this.subscriptions`
`this.quota`
`this.plans`

Then you don't have to worry about setting this flag... It's probably cleaner this way.

Reply:
Alternatively, we store all the requests and could do a Promise.all on those, and set ```this.billingError = true;``` in the `then` block? That might be a nice solution.

Reply:
ya could do. Sounds sensible

Reply:
Actually I think you mean the catch block, since you only want to flag it if there's error